numpy/random/generator.pyx

```diff
@@ -451,25 +452,29 @@ cdef class Generator:
         # bounded uniform integers. Lemire's method is preferable since it is
         # faster. randomgen allows a choice, we will always use the faster one.
         cdef bint _masked = False
-
-        if key == 'int32':
-            ret = _rand_int32(low, high, size, _masked, endpoint, &self._bitgen, self.lock)
-        elif key == 'int64':
-            ret = _rand_int64(low, high, size, _masked, endpoint, &self._bitgen, self.lock)
-        elif key == 'int16':
-            ret = _rand_int16(low, high, size, _masked, endpoint, &self._bitgen, self.lock)
-        elif key == 'int8':
-            ret = _rand_int8(low, high, size, _masked, endpoint, &self._bitgen, self.lock)
-        elif key == 'uint64':
-            ret = _rand_uint64(low, high, size, _masked, endpoint, &self._bitgen, self.lock)
-        elif key == 'uint32':
-            ret = _rand_uint32(low, high, size, _masked, endpoint, &self._bitgen, self.lock)
-        elif key == 'uint16':
-            ret = _rand_uint16(low, high, size, _masked, endpoint, &self._bitgen, self.lock)
-        elif key == 'uint8':
-            ret = _rand_uint8(low, high, size, _masked, endpoint, &self._bitgen, self.lock)
-        elif key == 'bool':
-            ret = _rand_bool(low, high, size, _masked, endpoint, &self._bitgen, self.lock)
+        try:
+            if key == 'int32':
+                ret = _rand_int32(low, high, size, _masked, endpoint, &self._bitgen, self.lock)
+            elif key == 'int64':
+                ret = _rand_int64(low, high, size, _masked, endpoint, &self._bitgen, self.lock)
+            elif key == 'int16':
+                ret = _rand_int16(low, high, size, _masked, endpoint, &self._bitgen, self.lock)
+            elif key == 'int8':
+                ret = _rand_int8(low, high, size, _masked, endpoint, &self._bitgen, self.lock)
+            elif key == 'uint64':
+                ret = _rand_uint64(low, high, size, _masked, endpoint, &self._bitgen, self.lock)
+            elif key == 'uint32':
+                ret = _rand_uint32(low, high, size, _masked, endpoint, &self._bitgen, self.lock)
+            elif key == 'uint16':
+                ret = _rand_uint16(low, high, size, _masked, endpoint, &self._bitgen, self.lock)
+            elif key == 'uint8':
+                ret = _rand_uint8(low, high, size, _masked, endpoint, &self._bitgen, self.lock)
+            elif key == 'bool':
+                ret = _rand_bool(low, high, size, _masked, endpoint, &self._bitgen, self.lock)
+        except ValueError as e:
+            if high_and_low_swapped and 'low >= high' == str(e):
+                e.args = ("low must be greater than 0 when high is not given.", )
```

Comment:
I'm not sure if this works correctly in displayed stack traces. What's the trade-off between doing this and raising a brand new exception `from None`?

Reply:
Yes, the stack trace is good. As an example, the code 
```
import numpy as np
rng = np.random.default_rng()
rng.integers(0)
```
produces the stacktrace 
```
Traceback (most recent call last):
  File "/Users/maxwellaladago/Documents/pub/numpy-max/perms.py", line 4, in <module>
    rng.integers(0)
  File "generator.pyx", line 458, in numpy.random.generator.Generator.integers
  File "bounded_integers.pyx", line 1228, in numpy.random.bounded_integers._rand_int64
ValueError: low must be greater than 0 when high is not given.
```

The problem with the `raise from None` is that it contains the phrase `raise from None` which can be confusing. Additionally, it doesn't capture the trace of the original exception. For example, 

```
import numpy as np
try:
    rng = np.random.default_rng()
    rng.integers(0)
except ValueError as e:
    raise ValueError("just a demo") from None

```
will produce the stacktrace 

```
Traceback (most recent call last):
  File "/Users/maxwellaladago/Documents/pub/numpy-max/perms.py", line 6, in <module>
    raise ValueError("just a demo") from None
ValueError: just a demo
```

Reply:
How about just preempting the whole thing and raising the error early on? That seems simpler to parse to me?